Documentation/Basic-Usage.md

```diff
@@ -47,57 +61,88 @@ struct User {
+
+[functional_concepts]: https://github.com/thoughtbot/Argo/blob/master/Documentation/Functional-Concepts.md
 
 ```swift
-return curry(Model.init) <^> paramOne <*> paramTwo <*> paramThree
+  static func decode(json: JSON) -> Decoded<DecodedType> {
+    return curry(Model.init) <^> paramOne <*> paramTwo <*> paramThree
+  }
 ```
 
-and so on. If any of those parameters are an error, the entire creation process
-will fail, and the function will return the first error. If all of the
-parameters are successful, the value will be unwrapped and passed to the
-`init` function.
+and so on. If any of those parameters are a failure state, then the entire
+creation process will fail, and the function will return the first failure
+state. If all of the parameters are successful, the value will be unwrapped and
+passed to the `init` function.
+
+### Safely pulling values from JSON
 
-In order to help with the decoding process, Argo introduces two new operators
-for parsing a value out of the JSON:
+In the example above, we showed some non-existent parameters (`paramOne`, etc), but
+one of Argo's main features is the ability to help you grab the real parameters
+from the JSON structure in a way that is type-safe and concise. You don't need
+to manually check to make sure that a value is non-nil, or that it's of the
+right type. Argo leverages Swift's expressive type system to do that leaving
+lifting for you. To help with the decoding process, Argo introduces two new
+operators for parsing a value out of the JSON:
 
 - `<|` will attempt to parse a single value from the JSON
 - `<||` will attempt to parse an array of values from the JSON
 
-The usage of these operators is the same regardless:
+The usage of these operators is simple:
+
+- `json <| "name"` is analogous to `json["name"]`
+- `json <|| "posts"` is analogous to `json["posts"]`
 
-- `json <| "key"` is analogous to `json["key"]`
-- `json <| ["key", "nested"]` is analogous to `json["key"]["nested"]`
+As a bonus, if your JSON contains nested data whose intermediate elements are
+also `Decodable`, then you can retrieve a nested value by using an array of
+strings:
 
-Both operators will attempt to parse the value from the JSON and will also
-attempt to cast the value to the expected type. If it can't find a value, the
-function will return a `Decoded.MissingKey(message: String)` error. If the
-value it finds is the wrong type, the function will return a
-`Decoded.TypeMismatch(message: String)` error.
+- `json <| ["location", "city"]` is analogous to `json["location"]["city"]`
+
+Each of these operators will attempt to extract the specified value from the
+JSON structure and if a value is found, the operator will also attempt to cast
+the value to the expected type. If it can't find a value, the function will
+return a `Decoded.MissingKey(message: String)` failure state. If the value it
+finds is of the wrong type, the function will return a
+`Decoded.TypeMismatch(message: String)` failure state.
```

Comment:
These are out of date. The `Decoded` type only has `Success` and `Failure` cases now. The `Failure` case can be `DecodeError.MissingKey(key: String)` or `DecodeError.TypeMismatch(expected: String, actual: String)`.